### PR TITLE
Feature/popup/confirm payment UI

### PR DESCRIPTION
**Pull request by chris-aeviator**
## Confirming payment
image

### Update
image

## Saving budget
image

### CODE REVIEW

File: src/common/utils/helpers.js [hunk 1]
```diff
@@ -29,5 +29,24 @@ export const getFiatFromSatoshi = async (currency, satoshi) => {
 };
 
 export const calcFiatFromSatoshi = (exchangeRate, satoshi) => {
```
Comment (bumi) [inline src/common/utils/helpers.js:31]: wondering if we should use obeject parameters here which might be more descriptive?

`calcFiatFromSatoshi({rate: xxx, satoshi: xxx})`

File: src/app/components/Prompt/Prompt.jsx [hunk 2]
```diff
@@ -18,13 +20,15 @@ class Prompt extends React.Component {
     const message = qs.parse(window.location.search);
     let origin = {};
     let args = {};
+    console.log(message)
     if (message.origin) {
       origin = JSON.parse(message.origin);
     }
     if (message.args) {
       args = JSON.parse(message.args);
+      var decoded = invoice.decode(args.paymentRequest)
```
[the remark this answers is not in the document]
Comment (bumi) [reply src/app/components/Prompt/Prompt.jsx:29]: and we can not rely on that there is a `args.paymentRequest` available. 
The Prompt component starts any user prompt, which is also `enable` for example and we will have more in the future that also have args (e.g. keysend)

**Review by bumi — COMMENTED** [review]
I think that bolt11.js file is not used?

File: src/app/components/Prompt/Prompt.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chris-aeviator) [reply src/app/components/Prompt/Prompt.jsx:29]: so we could check if `invoice.decode(args?.paymentRequest)` is defined.

File: src/common/utils/helpers.js (hunk 1 above)
Comment (chris-aeviator) [reply src/common/utils/helpers.js:31]: I'd do this on more parameters but not on two - up to you

**Comment by chris-aeviator** [comment]
> I think that bolt11.js file is not used?

Forgot to remove it

Github delete file is not working from the UI, will do a commit later
grafik

File: src/app/components/Prompt/Prompt.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chris-aeviator) [reply src/app/components/Prompt/Prompt.jsx:29]: Yeah checked the code

```
  var decodedPaymentRequest = invoice.decode(args?.paymentRequest)
    }
    this.state = { origin, args, type: message.type, invoice: decodedPaymentRequest ?? {}  };
```

should always do the job

[the remark this answers is not in the document]
Comment (bumi) [reply src/app/components/Prompt/Prompt.jsx:29]: maybe something like this? then we do not call the `decode` with null or something we know might not work
```
let decodedPaymentRequest = args.paymentRequest ? invoice.decode(args.paymentRequest) : {}
```

File: src/common/utils/helpers.js (hunk 1 above)
Comment (bumi) [reply src/common/utils/helpers.js:31]: yeah, don't mind now. just a thought, generally I prefer that better as it makes it easier for people to read.

**Comment by chris-aeviator** [comment]
@bumi is there anything left to be done on this?

**Comment by bumi** [comment]
ah, I think I missed your updates. I guess then feel free to merge it.

**Pull request merged**